packages/next/src/index.ts

```diff
@@ -2595,18 +2609,39 @@ async function getServerlessPages(params: {
   pagesDir: string;
   entryPath: string;
   outputDirectory: string;
+  appPathRoutesManifest?: Record<string, string>;
 }) {
-  const [pages, middlewareManifest] = await Promise.all([
+  const [pages, appPaths, middlewareManifest] = await Promise.all([
     glob('**/!(_middleware).js', params.pagesDir),
+    params.appPathRoutesManifest
+      ? glob('**/page.js', path.join(params.pagesDir, '../app'))
+      : Promise.resolve({}),
     getMiddlewareManifest(params.entryPath, params.outputDirectory),
   ]);
 
+  const normalizedAppPaths: typeof appPaths = {};
+
+  if (params.appPathRoutesManifest) {
+    for (const entry of Object.keys(params.appPathRoutesManifest || {})) {
+      const normalizedPath = `${path.join(
+        '.',
+        params.appPathRoutesManifest[entry]
+      )}.js`;
+      const globPath = `${path.join('.', entry)}.js`;
+
+      if (appPaths[globPath]) {
+        normalizedAppPaths[normalizedPath] = appPaths[globPath];
+      }
+    }
+  }
+
   // Edge Functions do not consider as Serverless Functions
   for (const edgeFunctionFile of Object.keys(
     middlewareManifest?.functions ?? {}
   )) {
+    delete normalizedAppPaths[edgeFunctionFile.slice(1) + '.js'];
```

Comment:
Nit pick: use an variable like 
```js
const edgePath = edgeFunctionFile.slice(1) + '.js';
delete normalizedAppPaths[edgePath];
delete pages[edgePath];
```

Reply:
Updated in https://github.com/vercel/vercel/pull/8098/commits/8953047b143e9134b30a4b9f12acaabada795af0